google/cloud/firestore_v1/_helpers.py

```diff
@@ -196,9 +196,12 @@ def encode_value(value) -> types.document.Value:
 
     # NOTE: We avoid doing an isinstance() check for a Document
     #       here to avoid import cycles.
-    document_path = getattr(value, "_document_path", None)
-    if document_path is not None:
-        return document.Value(reference_value=document_path)
+    # Add check to see if the object has a _document_path attribute
+    # to avoid getting a KeyError on embeded objects
+    if hasattr(value, "_document_path"):
```

Comment:
This cannot be the right fix for your issue:  the three-argument form of `getattr` returns a default value if the object does not have the given attribute.